src/adapters/scene_category.js

```diff
@@ -1,21 +1,63 @@
 import { Marker } from 'mapbox-gl--ENV';
 import constants from '../../config/constants.yml';
-import { createIcon } from '../adapters/icon_manager';
 import Telemetry from 'src/libs/telemetry';
 import { toUrl } from 'src/libs/pois';
 import { fire, listen } from 'src/libs/customEvents';
+import { poisToGeoJSON, emptyFeatureCollection } from 'src/libs/geojson';
+import { filteredPoisStyle } from 'src/adapters/pois_styles';
+import { createMapGLIcon, createIcon } from 'src/adapters/icon_manager';
 
-function getMarkerId(poi) {
-  return `marker_${poi.id}`;
-}
+const DYNAMIC_POIS_LAYER = 'poi-filtered';
 
 export default class SceneCategory {
   constructor(map) {
     this.map = map;
-    this.markers = [];
+    this.hoveredMarker = new Marker({
+      element: createIcon({ disablePointerEvents: true }),
+      anchor: 'bottom',
+    });
+    this.selectedMarker = new Marker({
+      element: createIcon({}),
+      anchor: 'bottom',
+    });
+
+    createMapGLIcon('./statics/images/map/pin_map_dot.svg', 50, 60)
+      .then(imageData => {
+        this.map.addImage('pin_with_dot', imageData);
+      });
+
+    this.map.addSource(DYNAMIC_POIS_LAYER, {
+      type: 'geojson',
+      data: emptyFeatureCollection,
+      promoteId: 'id', // tells MapBox-GL to use this property as internal feature identifier
+    });
+    this.map.addLayer({
+      ...filteredPoisStyle,
+      source: DYNAMIC_POIS_LAYER,
+      id: DYNAMIC_POIS_LAYER,
+    });
+    this.map.on('click', DYNAMIC_POIS_LAYER, e => {
+      e.originalEvent.stopPropagation();
+      const poi = this.getPointedPoi(e);
+      this.selectPoi({
+        poi,
+        pois: this.pois,
+        poiFilters: this.poiFilters,
+      });
+    });
```

Comment:
Possible edge-case here:
when several POIs are very close from each other and the mouse moved to another poi after hover (without triggering `mouseleave`), the clicked `poi` (from the dynamic layer) may be different from the highlighted one which is described in the current popup.

An alternative would be to always select the highlighted poi here, independently from `e` ? With potentially other bugs ?

Reply:
Haha, interesting edge case, you're always good with finding them :D
How I understand it, it's more a problem with the hovered POI (and therefore the popup) not being correcly updated than a problem with the clicked one, right?
![Peek 14-09-2020 18-07](https://user-images.githubusercontent.com/243653/93110139-37913c80-f6b5-11ea-8c0d-32e194de0390.gif)

Reply:
You're right. Ideally the highlighted poi and the popup would always follow the mouse, to be consistent the feature triggered on click.
But that's probably not feasible with only `mouseenter`/`mouseleave` events. So I suppose we would need to listen to `mousemove` events and introduce more complexity; I am not sure it's worth it.

Reply:
@amatissart @xem I gave it a try with `mousemove`. For the user it works by just replacing `mouseenter` by it. But it triggers invisible re-renders on each mouse movement, so I checked the POI actually changed before touching the marker or popup. I think it's not too complex after all.